.github/workflows/auto-pr.yml

```
@@ -10,71 +10,130 @@ jobs:
-        run: |
-          gh issue comment ${{ github.event.issue.number }} --body "Hi @${{ github.actor }}!
 
-          Thanks for your submission! Unfortunately, our automated test system found a few issues with your extension:
+            const { error, extensionName } = await extractExtension();
+
+            if(error === "zip-error")
+              github.rest.issues.createComment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: `👋 Thanks for your submission! Sadly, we were unable to properly extract your extension from it's zip file. Try to create the zip with another software, and to update your original submission post with the new file. If that still doesn't work, ask a member of the extension team for help 😉`
+              })
+
+            if(error === "too-many-files")
+              github.rest.issues.createComment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: `👋 Thanks for your submission! We are kind of confused; Your zip file contained more than one JSON file, and only one extension can be submitted at a time. Please update your original submission post with a new zip file containing only one extension file 🙏`
+              })
+
+            if(error === "no-json-found")
+              github.rest.issues.createComment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: `👋 Thanks for your submission! We are kind of confused; Your zip file doesn't seem to contain an extension. Please update your original submission post with a new zip file containing an extension 🙏\nIf your zip does already contain an extension, make sure your file ends with ".json", and that it is at the root of your zip file (not in a sub folder!), as otherwise it will be unrecognized and ignored by the system. If you've checked and nothing seems wrong with your zip, you can ask a member of the extension team for help 😉`
+              })
+
+            if(error === "invalid-file-name")
+              github.rest.issues.createComment({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: `👋 Thanks for your submission! We are sorry, but the filename of the extension has unrecognized characters. Since filenames with weird character can cause all kinds of problems and security flaws, our system won't allow file names with other characters than normal latin upper- and lowercase characters or numbers. Additionally, the first character must be an uppercase character. Please update your original submission post with a new zip file containing your extensions with another file name following those guidelines 🙏`
```

```suggestion
                body: `👋 Thanks for your submission! We are sorry, but the filename of the extension has unrecognized characters. Since filenames with weird character can cause all kinds of problems and security flaws, our system won't allow file names with other characters than normal latin upper- and lowercase characters or numbers. Additionally, the first character must be an uppercase character. Please update your original submission post with a new zip file containing your extension with another file name following those guidelines 🙏`
```

This message is duplicated, is this normal?

---

Yes, because both are basically the same error. That check is technically duplicated, but the first one is made with the file in JSZip and the other on the extracted file, so I prefer to keep both just in case there is a path name representation inconsistency between the system FS and the JSZip FS that would cause this to be somehow different